src/Mod/Sketcher/Gui/DrawSketchHandlerBSplineByInterpolation.h

```diff
@@ -0,0 +1,606 @@
+            for (size_t i = 1; i < BSplineKnots.size()-1; ++i) {
+                if (BSplineMults[i-1] >= myDegree && BSplineMults[i+1] >= myDegree)
+                    isBetweenC0Points[i] = true;
+            }
+
+            int currentgeoid = getHighestCurveIndex();
+
+            try {
+                //Gui::Command::openCommand(QT_TRANSLATE_NOOP("Command", "Add B-spline curve"));
+
+                // TODO: Bypass this for when there are no C0 knots
+                // TODO: Create B-spline in pieces between C0 knots
+                Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_poles = []");
+                Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_knots = []");
+                Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_mults = []");
+                Gui::Command::runCommand(Gui::Command::Gui, "_bsps = []");
+                for (auto & controlpoints: controlpointses) {
+                    // FIXME: can we get by without naming the variable?
+                    // TODO: variable degrees?
+                    QString cmdstr = QString::fromLatin1("_bsps.append(Part.BSplineCurve())\n"
+                                                         "_bsps[-1].interpolate(%1, PeriodicFlag=%2)\n"
+                                                         "_bsps[-1].increaseDegree(%3)")
+                        .arg(QString::fromLatin1(controlpoints.c_str()))
+                        .arg(QString::fromLatin1(ConstrMethod == 0 ?"False":"True"))
+                        .arg(myDegree);
+                    Gui::Command::runCommand(Gui::Command::Gui, cmdstr.toLatin1());
+                    // TODO: Adjust internal knots here (raise multiplicity)
+                    // How this contributes to the final B-spline
+                    if (controlpoints == controlpointses.front()) {
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_poles.extend(_bsps[-1].getPoles())");
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_knots.extend(_bsps[-1].getKnots())");
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_mults.extend(_bsps[-1].getMultiplicities())");
+                    }
+                    else {
+                        // TODO: Adjust for periodic
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_poles.extend(_bsps[-1].getPoles()[1:])");
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_knots.extend([_finalbsp_knots[-1] + i for i in _bsps[-1].getKnots()[1:]])");
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_mults[-1] = 3"); // FIXME: Hardcoded
+                        Gui::Command::runCommand(Gui::Command::Gui, "_finalbsp_mults.extend(_bsps[-1].getMultiplicities()[1:])");
+                    }
+                }
+
+                // TODO: Join the pieces of the B-splines

```

Comment:
Could you just search for TODO in this file and check that those are actually TODOs at this stage?

Reply:
Sorry for tiring you out.

Reply:
No problem. You are methodical and that is very good. I am just doing a review and try to think the information the future developer reading the code needs. The TODOs that are yet to be done must remain. The TODOS that are done will be confusing for that future developer.

Reply:
TODOs no longer necessary are removed.